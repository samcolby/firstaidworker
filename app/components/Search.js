import React, { PureComponent } from "react";
import PropTypes from "prop-types";

import { Platform, StyleSheet } from "react-native";
import { SearchBar } from "react-native-elements";

import { COLORS } from "../Constants";

/**
 * React Pure Component for a search input control
 *
 * @class Search
 * @extends {PureComponent}
 */
class Search extends PureComponent {
  static propTypes = {
    onCancel: PropTypes.func,
    onChangeText: PropTypes.func,
    onClear: PropTypes.func,
    value: PropTypes.string
  };

  constructor(props) {
    super(props);
    this.state = { query: props.value };
  }

  onChangeText = query => {
    this.setState({ query });
    if (typeof this.props.onChangeText === "function") {
      this.props.onChangeText(query);
    }
  };

  render() {
    return (
      <SearchBar
        containerStyle={styles.containerStyle}
        onCancel={this.props.onCancel}
        onChangeText={this.onChangeText}
        onClear={this.props.onClear}
        placeholder="Search..."
        platform={Platform.OS}
        value={this.state.query}
      />
    );
  }
}

const styles = StyleSheet.create({
  containerStyle: {
    backgroundColor: COLORS.BACKGROUND
  },
  inputStyle: {
    backgroundColor: "#eee"
  }
});

export default Search;
