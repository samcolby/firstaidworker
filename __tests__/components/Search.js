// __tests__/Intro-test.js
import React from "react";

import { shallow } from "enzyme";

import Search from "../../app/components/Search";

describe("<Search /> component", () => {
  it("renders correctly", () => {
    const component = shallow(<Search />);

    expect(component.length).toBe(1);
    expect(component).toMatchSnapshot();
  });

  it("handles the value prop", () => {
    const testQuery = "Default query";
    const component = shallow(<Search value={testQuery} />);

    expect(component.prop("value")).toEqual(testQuery);
    expect(component).toMatchSnapshot();
  });

  it("handles onChangeText", () => {
    const onChangeText = jest.fn();
    const component = shallow(<Search onChangeText={onChangeText} />);

    component.simulate("changeText");
    expect(onChangeText).toHaveBeenCalled();
    expect(component).toMatchSnapshot();
  });

  it("returns the correct onChangeText value", () => {
    const onChangeText = jest.fn();
    const testQuery = "My Query Is";

    const component = shallow(<Search onChangeText={onChangeText} />);

    component.simulate("changeText", testQuery);

    expect(onChangeText).toHaveBeenCalledWith(testQuery);
    expect(component).toMatchSnapshot();
  });

  it("handles onClear", () => {
    const onClear = jest.fn();
    const component = shallow(<Search onClear={onClear} />);

    component.simulate("clear");
    expect(onClear).toHaveBeenCalled();
    expect(component).toMatchSnapshot();
  });

  it("handles onCancel", () => {
    const onCancel = jest.fn();
    const component = shallow(<Search onCancel={onCancel} />);

    component.simulate("cancel");
    expect(onCancel).toHaveBeenCalled();
    expect(component).toMatchSnapshot();
  });
});
