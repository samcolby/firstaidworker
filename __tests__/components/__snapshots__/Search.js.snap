// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> component handles onCancel 1`] = `
<ForwardRef(Themed.SearchBar)
  containerStyle={
    Object {
      "backgroundColor": "#fff",
    }
  }
  onCancel={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  onChangeText={[Function]}
  placeholder="Search..."
  platform="ios"
  value=""
/>
`;

exports[`<Search /> component handles onChangeText 1`] = `
<ForwardRef(Themed.SearchBar)
  containerStyle={
    Object {
      "backgroundColor": "#fff",
    }
  }
  onChangeText={[Function]}
  placeholder="Search..."
  platform="ios"
/>
`;

exports[`<Search /> component handles onClear 1`] = `
<ForwardRef(Themed.SearchBar)
  containerStyle={
    Object {
      "backgroundColor": "#fff",
    }
  }
  onChangeText={[Function]}
  onClear={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  placeholder="Search..."
  platform="ios"
  value=""
/>
`;

exports[`<Search /> component renders correctly 1`] = `
<ForwardRef(Themed.SearchBar)
  containerStyle={
    Object {
      "backgroundColor": "#fff",
    }
  }
  onChangeText={[Function]}
  placeholder="Search..."
  platform="ios"
  value=""
/>
`;

exports[`<Search /> component returns the correct onChangeText value 1`] = `
<ForwardRef(Themed.SearchBar)
  containerStyle={
    Object {
      "backgroundColor": "#fff",
    }
  }
  onChangeText={[Function]}
  placeholder="Search..."
  platform="ios"
  value="My Query Is"
/>
`;
